maintainers/gpuci.md

```diff
@@ -0,0 +1,247 @@
+---
+layout: default
+nav_order: 4
+parent: RAPIDS Maintainer Docs
+title: gpuCI
+---
+
+# gpuCI Technical Documentation
+
+## Overview
+
+Outlines how gpuCI is configured for RAPIDS and how to integrate other projects
+
+gpuCI is accessible here: https://gpuci.gpuopenanalytics.com/
+
+### Intended audience
+
+Developers
+{: .label .label-green}
+
+Project Leads
+{: .label .label-blue}
+
+Operations
+{: .label .label-purple}
+
+## Infrastructure
+
+This is an overview of the tools used in gpuCI
+
+### Jenkins
+
+- Open source project for CI/CD and general automation that is very extensible and well adopted
+- gpuCI
+  - Build scripts are standardized across RAPIDS projects
+    - CPU builds with conda & pip
+    - GPU builds & unit tests
+    - Executing style & changelog checks
+    - Planned to be open sourced soon
+  - Standardization allows for scripts to create and config jobs
+    - [https://gpuci.gpuopenanalytics.com/view/all/job/Rapids Seed Job/](https://gpuci.gpuopenanalytics.com/view/all/job/Rapids%20Seed%20Job/)
+    - Uses Jenkins job-dsl plugin
+    - Creates 125+ Jenkins jobs to perform everything listed in this document
+    - Handles the links between Jenkins & GitHub
+  - Custom [plugin](https://github.com/gpuopenanalytics/remote-docker-plugin) written & open sourced by RAPIDS to better handle GPU testing with nvidia-docker
+
+### Docker containers
+
+All jobs which use Docker use the `gpuci/rapidsai-base` images.
+
+[Docker Hub](https://hub.docker.com/r/gpuci/rapidsai-base)
+
+[GitHub](https://github.com/rapidsai/gpuci-build-environment)
+
+#### Conda environment
+
+The images contain a common conda environment with basic tools such as `python`, `cmake`, `make`, `cython`, and `flake8`. See the full list [here](https://github.com/rapidsai/gpuci-build-environment/blob/master/Dockerfile#L66)
+
+The conda environment can be used in the scripts with `source activate gdf`
+
+## Jobs
+
+### Pull requests
+
+Pull requests trigger test builds using [GitHub pull request builder plugin](https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin).
+
+Each PR is build with four types of jobs.
+
+#### Changelog
+
+Uses the `ci/checks/changelog.sh` script to determine success or failure
+
+#### Style
+
+Uses the `ci/checks/style.sh` script to determine success or failure
+
+#### CPU builds
+
+Each PR is built in a container with a matrix of parameters:
+- CUDA 9.2 & 10.0
+- Python 3.6 & 3.7
+
+This allows for testing compilation against multiple CUDA and Python versions.
+
+#### GPU build
+
+Each PR is built in a container using CUDA 10.0 and Python 3.6. the purpose of this job is primarily for testing with a GPU present.
```

Comment:
Is there a historical reason this is called "GPU build" even though the primary purpose is testing? It might be worth mentioning for context, or to emphasize that new projects *shouldn't* build on GPU systems as a best practice.